Sources/Confidence/Confidence.swift

```diff
@@ -238,28 +219,59 @@ public class Confidence: ConfidenceEventSender {
         return reconciledCtx
     }
 
-    public func putContext(key: String, value: ConfidenceValue) {
-        withLock { confidence in
+    public func putContext(key: String, value: ConfidenceValue) async {
+        await withLockAsync { confidence in
             var map = confidence.contextSubject.value
             map[key] = value
             confidence.contextSubject.value = map
-            confidence.debugLogger?.logContext(action: "PutContext", context: confidence.contextSubject.value)
+            do {
+                try await self.fetchAndActivate()
+                confidence.debugLogger?.logContext(action: "PutContext", context: confidence.contextSubject.value)
+            } catch {
+                confidence.debugLogger?.logMessage(message: "Error when putting context: \(error)", isWarning: true)
+            }
         }
     }
 
-    public func putContext(context: ConfidenceStruct) {
+
+    public func putContextLocal(context: ConfidenceStruct, removeKeys removedKeys: [String] = []) {
```

Comment:
Just a draft API (not in our internal specs yet). This is to experiment how to solve the integration with the OF layer